src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -218,7 +227,22 @@ public Explanation explain(LeafReaderContext context, int doc) throws IOExceptio
                     subs.add(Explanation.match(explain.getValue(), featureString, explain));
                     d.setFeatureScore(ordinal, explain.getValue());
                 }
+
+                // Add feature binding, bound to name if set, otherwise $[ordinal]
+                Feature f = features.feature(ordinal);
+                bindings.add(f.name() == null ? ("$" + ordinal) : f.name(),
```

Comment:
you removed the `$ord` syntax
Also it might be faster/easier to read to create an anonymous class on Bindings reusing most of the code you wrote in `scorer`